### PR TITLE
Improve output readability for `mc stat` command

### PR DESCRIPTION
**Pull request by r-scheele**
Description

This PR adjusts the output of the mc stat command to display the server-side encryption information in a more understandable manner.

Motivation and Context

This PR aims to aims to adjust the output from:
```
>mc stat -r play/abdul
Name      : abdul/kms
Date      : 2023-06-12 06:58:38 PDT
Size      : 27 B
ETag      : d40ed931ae5772ee918abb5fb18d0ad2
Type      : file
Metadata  :
  Content-Type: application/octet-stream
Encrypted :
  X-Amz-Server-Side-Encryption-Aws-Kms-Key-Id: arn:aws:kms:my-minio-key
  X-Amz-Server-Side-Encryption               : aws:kms

Name      : abdul/s3
Date      : 2023-06-12 06:58:53 PDT
Size      : 27 B
ETag      : 5848d6f7d2172580f07b1ed8d7b56e9a
Type      : file
Metadata  :
  Content-Type: application/octet-stream
Encrypted :
  X-Amz-Server-Side-Encryption: AES256

Name      : abdul/unencrypted
Date      : 2023-06-12 06:58:02 PDT
Size      : 27 B
ETag      : 5848d6f7d2172580f07b1ed8d7b56e9a
Type      : file
Metadata  :
  Content-Type: application/octet-stream


>mc stat -r play/abdul --encrypt-key "play/abdul=MzJieXRlc2xvbmdzZWNyZWFiY2RlZmcJZ2l2ZW5uMjE="
Name      : abdul/sse-c
Date      : 2023-06-12 07:00:33 PDT
Size      : 27 B
ETag      : 77f1f1f52aac4f62e57b8e6e973ee603
Type      : file
Metadata  :
  Content-Type: application/octet-stream
Encrypted :
  X-Amz-Server-Side-Encryption-Customer-Key-Md5  : i5MnXlnfBFzo+X31rRClNw==
  X-Amz-Server-Side-Encryption-Customer-Algorithm: AES256
```

To this:
```
$ ./mc stat -r play/abdul --encrypt-key "play/abdul=MzJieXRlc2xvbmdzZWNyZWFiY2RlZmcJZ2l2ZW5uMjE="
Name      : abdul/test2
Date      : 2023-06-13 10:42:34 PDT 
Size      : 0 B    
ETag      : 982b9d404db37a97dc7c9f49bb312cc7 
Type      : file 
Encryption: SSE-C
Metadata  :
  Content-Type: application/octet-stream

$ ./mc stat -r play/abdul 
Name      : abdul/test
Date      : 2023-06-13 10:29:13 PDT 
Size      : 0 B    
ETag      : d41d8cd98f00b204e9800998ecf8427e 
Type      : file 
Encryption: SSE-S3
Metadata  :
  Content-Type: application/octet-stream 

Name      : abdul/test1
Date      : 2023-06-13 10:29:56 PDT 
Size      : 0 B    
ETag      : 79204d0af4ecf4741d48f10c0e9a690b 
Type      : file 
Encryption: SSE-KMS (arn:aws:kms:my-minio-key)
Metadata  :
  Content-Type: application/octet-stream 
```

How to test this PR?
- Checkout this branch locally.
- Build the modified mc client.
- Run the following commands to test the mc stat output:
  `./mc stat play/abdul`
  `./mc stat -r play/abdul --encrypt-key "play/abdul=MzJieXRlc2xvbmdzZWNyZWFiY2RlZmcJZ2l2ZW5uMjE="` 

Types of changes:
 New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by r-scheele** [comment]
Kannappan asked me to remove the header, but I forgot to add the kms keyId, I'll add it to my next commit.

**Comment by klauspost** [comment]
```
mc: test_mv_object: /home/runner/work/mc/mc/mc --config-dir /tmp/.mc-9677 --quiet --no-color stat myminio/mc-test-bucket-364/dir-15742-6196/object-1  >>> 
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/minio/mc/cmd.statMessage.String({{0x0, 0x0}, {0xc00004cd08, 0x14}, {0x0, 0xedc197269, 0x25c2600}, 0x100000, {0xc000026181, 0x20}, ...})
	github.com/minio/mc/cmd/stat.go:120 +0x1165
github.com/minio/mc/cmd.printMsg({0x1ca28f8, 0xc000001080})
	github.com/minio/mc/cmd/print.go:38 +0x192
github.com/minio/mc/cmd.statURL({0x1ca78f8, 0xc0006ec9b0}, {0x7fff826efd9c, 0x32}, {0x0, 0x0}, {0x4ab8e5?, 0x67e22c?, 0x0?}, 0x0, ...)
	github.com/minio/mc/cmd/stat.go:295 +0x1865
github.com/minio/mc/cmd.mainStat(0x4ab8e5?)
	github.com/minio/mc/cmd/stat-main.go:168 +0x528
github.com/minio/cli.HandleAction({0x181cbc0?, 0x1a982d0?}, 0x4?)
	github.com/minio/cli@v1.24.2/app.go:490 +0x99
github.com/minio/cli.Command.Run({{0x19e37c6, 0x4}, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0x19fb2f5, 0x14}, {0x0, ...}, ...}, ...)
	github.com/minio/cli@v1.24.2/command.go:242 +0xf0a
github.com/minio/cli.(*App).Run(0xc00042c700, {0xc000040070, 0x7, 0x7})
	github.com/minio/cli@v1.24.2/app.go:260 +0xfa5
github.com/minio/mc/cmd.Main({0xc000040070?, 0x7, 0x7})
	github.com/minio/mc/cmd/main.go:141 +0x645
main.main()
	github.com/minio/mc/main.go:28 +0x4b
```

**Review by harshavardhana — APPROVED** [review]

File: cmd/pipe_unsupported.go [hunk 1]
```diff
@@ -21,7 +21,12 @@ package cmd
 
 import "os"
 
-func increasePipeBufferSize(f *os.File, desiredPipeSize int) error {
+// func increasePipeBufferSize(f *os.File, desiredPipeSize int) error {
+// 	// this is not supported on non-Linux platforms.
+// 	return nil
+// }
```
Comment (donatello) [inline cmd/pipe_unsupported.go:27]: This commented function can be removed

**Review by donatello — CHANGES_REQUESTED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**